public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -134,33 +93,12 @@ export default class CloudMonitoringDatasource extends DataSourceApi<CloudMonito
       .toPromise();
   }
 
-  getTimeSeries(options: DataQueryRequest<CloudMonitoringQuery>): Observable<PostResponse> {
-    const queries = options.targets
-      .map(this.migrateQuery)
-      .filter(this.shouldRunQuery)
-      .map(q => this.prepareTimeSeriesQuery(q, options.scopedVars))
-      .map(q => ({ ...q, intervalMs: options.intervalMs, type: 'timeSeriesQuery' }));
-
-    if (!queries.length) {
-      return of({ results: [] });
-    }
-
-    return from(this.ensureGCEDefaultProject()).pipe(
-      mergeMap(() => {
-        return this.api.post({
-          from: options.range.from.valueOf().toString(),
-          to: options.range.to.valueOf().toString(),
-          queries,
-        });
-      }),
-      map(({ data }) => {
-        return data;
-      })
-    );
+  applyTemplateVariables(target: CloudMonitoringQuery, scopedVars: ScopedVars): Record<string, any> {
```

Comment:
Looks like you can move all the code from prepareTimeSeriesQuery and put it into this function.